### DISCUSSION TITLE
Using tailwind with storybook and nextjs

### DISCUSSION THREAD

**Discussion by Akshay090** [tailwindlabs/tailwindcss#2929] [Help] [answered]
I'm trying to use tailwind with nextjs + storybook, but am currently getting error 
```
Error: PostCSS plugin tailwindcss requires PostCSS 8.
```
I've even upgraded all packages with ```yarn upgrade --latest``` still it's not resolved. 

Here's a minimal git repo I created with this issue: https://github.com/Akshay090/next-tailwind-story

Please do suggest some approaches to solve this.

Thank you.

**Comment by alveshelio** [comment]
I'm having the same issue. I have the most up to date versions:
```
    "tailwindcss": "^2.0.1". 
    "postcss": "^8.1.10",
    "@storybook/addon-actions": "^6.1.8",
    "@storybook/addon-info": "^5.3.21",
    "@storybook/addon-knobs": "^6.1.8",
    "@storybook/addon-notes": "^5.3.21",
    "@storybook/addons": "^6.1.8",
    "@storybook/react": "^6.1.8",
```

**Comment by simonswiss** [comment, marked as answer]
Hey!

Some of the tooling within Storybook relies on PostCSS v7, and Tailwind 2.0 needs PostCSS v8 to work.

Because this is currently a recurring problem (a lot of tooling in the front end ecosystem still relies on PostCSS 7), we've created a `compat` version of Tailwind CSS that will work with PostCSS 7:

https://tailwindcss.com/docs/installation#post-css-7-compatibility-build

You should be able to use Tailwind with Storybook with that version of the package, until the dependencies involved in Storybook are upgraded to use PostCSS v8 👍

**Reply by alveshelio** [reply]
Thank you @simonswiss,

I've went ahead and followed the solution and it fixed my problem.

**Reply by mattiloh** [reply]
Hey! I bumped into the same issue and it turns out using the Tailwind `compat` version is not necessary anymore.

Instead, you can simply use the [Storybook PostCSS Addon](https://storybook.js.org/addons/@storybook/addon-postcss) with a slightly adjusted config to use the already installed PostCSS version (as described in the linked page at "PostCSS 8+").

**Reply by trexchard** [reply]
Hi @mattiloh! Do you have any details on how this works?

**Reply by mattiloh** [reply]
Sure! I simply followed the **PostCSS 8+** section in this guide (after adding the addon): https://storybook.js.org/addons/@storybook/addon-postcss

**Reply by jaotors** [reply]
@mattiloh hi, even though I add the @storybook/addon-postcss, i kept having error in PostCSS plugin tailwindcss requiress PostCSS 8, if i run the storybook

**Reply by kiliw** [reply]
@jaotors I had the same issue with the error 
``` 
PostCSS plugin tailwindcss requires PostCSS 8
```

It ended up being me writing the wrong config for the storybook, I hat the `addon-postcss` twice in my config.
```
module.exports = {
    stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(ts|tsx)'],
    addons: [
        '@storybook/addon-links',
        '@storybook/addon-essentials',
        '@storybook/addon-postcss' // THIS WAS THE MISTAKE
        {
            name: '@storybook/addon-postcss',
            options: {
                postcssLoaderOptions: {
                    implementation: require('postcss'),
                },
            },
        },
    ],
}
```

Maybe it helps

**Reply by tommaton** [reply]
@kiliw you've saved me a hell a lot of wasted time, I had it setup exactly this (incorrect) way... Made the change you suggested and now it works!

**Reply by kiliw** [reply]
Glad I could help

**Comment by Ridermansb** [comment]
For some reason.. after I followed the tutorial I get this error..

`TypeError: value.charCodeAt is not a function`

**Reply by coderinblack08** [reply]
yeah i get the same problem
